src/Config.js

```diff
@@ -100,6 +106,17 @@ module.exports = class extends ChainedMap {
               .forEach(name => this.plugin(name).merge(value[name]));
           }
 
+          case 'plugins': {
+            if (Array.isArray(value)) {
```

Comment:
yes it is intended since plugins can only be an array https://webpack.js.org/configuration/plugins/#plugins

though we could add a warning if it is not an array?